### PR TITLE
LPS-153046,LPS-152377

### PR DESCRIPTION
**Pull request by rebsilva**
Related Issues: 

- https://issues.liferay.com/browse/LPS-153046
-  https://issues.liferay.com/browse/LPS-152377

Hi Reviewer!

This PR is related to the 7.4 Polls Upgrade bugs, you can know more about them in the links above; I will leave here some advice that the PO said about these bugs:

_Attention: make sure we don't break anything in the customers that already are in 7.4 and are only using forms (for a missing flag for example)_

Thank you for your time!

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:relevant

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsIdToFormsIdUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+						portletPreferencesId, " AND name = 'questionId'"));
+			}
+		}
+	}
+
+	private static final String _OLD_DISPLAY_PORTLET_NAME =
+		"com_liferay_polls_web_portlet_PollsDisplayPortlet";
+
+	private static final String _OLD_PORTLET_NAME =
+		"com_liferay_polls_web_portlet_PollsPortlet";
```
[the remark this answers is not in the document]
Comment (rebsilva) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsIdToFormsIdUpgradeProcess.java:84]: Make sense, I will do that thanks!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsIdToFormsIdUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import com.liferay.portal.kernel.upgrade.BasePortletIdUpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Rebeca Silva
+ */
+public class PollsIdToFormsIdUpgradeProcess
+	extends BasePortletIdUpgradeProcess {
```
[the remark this answers is not in the document]
Comment (rebsilva) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsIdToFormsIdUpgradeProcess.java:28]: No need to apologize at all! I will try this name and let's see what happens

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsIdToFormsIdUpgradeProcess.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+				DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM
+			},
+			{_OLD_PORTLET_NAME, DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM_ADMIN}
+		};
+	}
+
+	private void _upgradePortletPreferenceValue() throws Exception {
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				StringBundler.concat(
+					"select portletPreferencesId, portletId from ",
```
[the remark this answers is not in the document]
Comment (rebsilva) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsIdToFormsIdUpgradeProcess.java:62]: You are right, I forgot to remove it, thanks a lot!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsIdToFormsIdUpgradeProcess.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+				"delete from Portlet where portletId = '",
+				_OLD_DISPLAY_PORTLET_NAME, "' OR portletId = '",
+				_OLD_PORTLET_NAME, "'"));
+		runSQL(
+			StringBundler.concat(
+				"delete from ResourcePermission where name = '",
+				_OLD_DISPLAY_PORTLET_NAME, "' OR name = '", _OLD_PORTLET_NAME,
+				"'"));
+
+		super.doUpgrade();
```
[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsIdToFormsIdUpgradeProcess.java:43]: It's fine for me, thanks for explaining!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsIdToFormsIdUpgradeProcess.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+			while (resultSet.next()) {
+				long portletPreferencesId = resultSet.getLong(
+					"portletPreferencesId");
+
+				runSQL(
+					StringBundler.concat(
+						"update PortletPreferenceValue set name = ",
+						"'formInstanceId' where portletPreferencesId=",
+						portletPreferencesId, " AND name = 'questionId'"));
```
[the remark this answers is not in the document]
Comment (rebsilva) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsIdToFormsIdUpgradeProcess.java:75]: I can try that and see if it works!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_0_0/PollsToDDMUpgradeProcess.java [hunk 6]
```diff
@@ -859,36 +879,37 @@ private void _upgradePollsQuestion(
...
+		DDMFormField ddmFormField = getDDMFormField(
+			ddmFormFieldOptions, description);
 
 		String definition = DDMFormSerializeUtil.serialize(
 			getDDMForm(ddmFormField), _ddmFormSerializer);
 
 		long structureId = _addDDMStructure(
 			groupId, companyId, userId, userName, createDate, modifiedDate,
-			name, description, definition, lastPublishDate);
+			name, "", definition, lastPublishDate);
```
[the remark this answers is not in the document]
Comment (rebsilva) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_0_0/PollsToDDMUpgradeProcess.java:890]: Sure, is better this way for sure, thanks!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_0_0/PollsToDDMUpgradeProcess.java [hunk 7]
```diff
@@ -844,6 +860,10 @@ private void _upgradeDDMFormInstance() throws Exception {
 		if (!hasColumn("DDMFormInstance", "expirationDate")) {
 			runSQL("alter table DDMFormInstance add expirationDate DATE null");
 		}
+
+		if (!hasColumn("DDMFormInstance", "convertedPolls")) {
+			runSQL("alter table DDMFormInstance add convertedPolls LONG");
+		}
```
[the remark this answers is not in the document]
Comment (rebsilva) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_0_0/PollsToDDMUpgradeProcess.java:866]: The ideal was to use boolean but I was having a hard time with the sintaxe, I will search again for examples in the code about this

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java [hunk 8]
```diff
@@ -197,8 +211,13 @@ else if (Objects.equals(
...
 
-				fieldValueJSONObject.put(
-					"value", String.valueOf(expirationDate != null));
+				if (convertedPolls == 1) {
+					fieldValueJSONObject.put("value", Boolean.TRUE.toString());
+				}
+				else {
+					fieldValueJSONObject.put(
+						"value", String.valueOf(expirationDate != null));
+				}
```
[the remark this answers is not in the document]
Comment (rebsilva) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java:220]: That's great!! I will try this, thanks!!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsIdToFormsIdUpgradeProcess.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (rebsilva) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsIdToFormsIdUpgradeProcess.java:43]: No problem!

**Comment by rebsilva** [comment]
ci:test:sf

**Comment by rebsilva** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsPortletIdToDDMPortletIdUpgradeProcess.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+		runSQL(
+			StringBundler.concat(
+				"delete from Portlet where portletId = '",
+				_PORTLET_ID_POLLS_DISPLAY, "' OR portletId = '",
+				_PORTLET_ID_POLLS, "'"));
+		runSQL(
+			StringBundler.concat(
+				"delete from ResourcePermission where name = '",
+				_PORTLET_ID_POLLS_DISPLAY, "' OR name = '", _PORTLET_ID_POLLS,
+				"'"));
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsPortletIdToDDMPortletIdUpgradeProcess.java:42]: @rebsilva could you please put the `_PORTLET_ID_POLLS_DISPLAY` after the `_PORTLET_ID_POLLS` to keep the alphabetical order?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsPortletIdToDDMPortletIdUpgradeProcess.java [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+	@Override
+	protected String[][] getRenamePortletIdsArray() {
+		return new String[][] {
+			{
+				_PORTLET_ID_POLLS_DISPLAY,
+				DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM
+			},
+			{_PORTLET_ID_POLLS, DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM_ADMIN}
+		};
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsPortletIdToDDMPortletIdUpgradeProcess.java:57]: @rebsilva same as above

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java [hunk 11]
```diff
@@ -216,10 +238,16 @@ else if (Objects.equals(
...
 
 				fieldValueJSONObject.put(
-					"value", String.valueOf(expirationDate != null));
+					"value", _getValue(convertedPolls, expirationDate));
 
 				hasShowPartialResultsToRespondents = true;
 			}
+			else if (Objects.equals(
+						fieldValueJSONObject.getString("name"), "published") &&
+					 convertedPolls) {
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_2/DDMFormInstanceUpgradeProcess.java:247]: @rebsilva could you please change to `convertedPolls && Objects.equals(fieldValueJSONObject.getString("name"), "published")`? Thanks!

**Review by carolmariaabb — CHANGES_REQUESTED** [review]
Amazing work, @rebsilva!!

Just a few (small) changes and we're good to go!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsPortletIdToDDMPortletIdUpgradeProcess.java (hunk 9 above)
Comment (rebsilva) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v5_1_4/PollsPortletIdToDDMPortletIdUpgradeProcess.java:42]: Sure!

**Comment by rebsilva** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by rebsilva** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:reevaluate:1357401_1564

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**